extensions/objects/templates/math-editor.component.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for math editor.
+ */
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+
+angular.module('oppia').component('mathEditor', {
+  bindings: {
+    value: '='
+  },
+  template: require('./math-editor.component.html'),
+  controller: [function() {
+    const ctrl = this;
+
+    ctrl.assignIdToGuppyDiv = function() {
+      // Dynamically assigns a unique id to the guppy-div
+      var divId = 'guppy_' + Math.floor(Math.random() * 100000000);
+      $('.guppy-div').attr('id', divId);
+      return divId;
+    };
+
+    ctrl.createGuppyInstance = function(divId: string) {
+      return new Guppy(divId, {});
+    };
+
+    ctrl.$onInit = function() {
```

Comment:
Let's say there are multiple components on a page. How many times is this called? Please test and verify.

If once per component, you'll probably want a common service instead, as suggested in my previous comment.

Reply:
I've tested this and yes, there can be, or actually there will be multiple guppy instances on a single page. But that would only happen when the creator saves a response and then adds another one (that would create a new guppy instance for each response). To handle this, we would have to fetch all the guppy divs and individually initialize them. But, this scenario wouldn't occur right now since we can't save a response atm(the interaction files haven't been added). The interaction files are added in my next [PR](https://github.com/oppia/oppia/pull/9553), and there I have modified the code to work where multiple instances are active on a single page.

Reply:
Please fix this issue properly in this PR (you will need to sort out the CI tests anyway). We don't merge PRs that are half done and need to be revisited later.

Reply:
Done.
RE the failing CI tests, I'd discussed this with @aks681 and it's most likely a flake. I've seen it fail on a couple of other PRs as well with the same exact error. Also, it's failing on the code which hasn't been modified in this PR.